joplin/base/views/joplin_views.py

```diff
@@ -0,0 +1,32 @@
+from django.shortcuts import get_object_or_404, redirect, render
+from wagtail.core.models import Page, UserPagePermissionsProxy
+from wagtail.admin.views import pages
+from wagtail.admin import messages
+from django.utils.translation import ugettext as _
+from django.urls import reverse
+
+def publish(request, page_id):
```

Comment:
This enables publishing from the index page with a view that is very similar to the "confirm unpublish" page

Reply:
I wasn't able to find any functionality in wagtail for publishing based on ID alone, it was based on submitting a page with all of the page data to that endpoint, which doesn't exist on the content list page.